### PR TITLE
[Merged by Bors] - chore(NumberTheory/Fermat): rename `fermat` to `Nat.fermatNumber`

### PR DESCRIPTION
**Pull request by Vierkantor**
(And also namespace and rename the lemmas about them.)

As discussed in the reviewers chatroom.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/NumberTheory/Fermat.lean [hunk 1]
```diff
@@ -12,59 +12,66 @@ import Mathlib.Tactic.Ring.RingNF
...
+  rw [fermatNumber_product, Nat.sub_add_cancel]
+  exact le_of_lt <| two_lt_fermatNumber _
 
 /--
 **Goldbach's theorem** : no two distinct Fermat numbers share a common factor greater than one.
 
 From a letter to Euler, see page 37 in [juskevic2022].
 -/
-theorem coprime_fermat_fermat {k n : ℕ} (h : k ≠ n) : Coprime (fermat n) (fermat k) := by
+theorem coprime_fermatNumberNumber_fermatNumber {k n : ℕ} (h : k ≠ n) :
```
Comment (Ruben-VandeVelde) [inline Mathlib/NumberTheory/Fermat.lean:66]: This doesn't look right

Comment (Vierkantor) [reply Mathlib/NumberTheory/Fermat.lean:66]: Oops, nicely spotted!
```suggestion
theorem coprime_fermatNumber_fermatNumber {k n : ℕ} (h : k ≠ n) :
```

**Comment by acmepjz** [comment]
I think `Nat.fermat` is enough, unless there are something else named "Fermat" for natural numbers. This makes the name `Nat.coprime_fermatNumber_fermatNumber` too long IMHO.

**Comment by vihdzp** [comment]
Agree, `Nat.fermat` seems fine unless there's some clashing with the FLT project or something like that.

**Comment by kbuzzard** [comment]
There's no clashing with FLT naming but I look at `Nat.fermat` and think "whatever does that mean?" whereas I look at `Nat.fermatNumber` and think "I know exactly what that will be" which is I think what we're aiming for.

File: Mathlib/NumberTheory/Fermat.lean [hunk 2]
```diff
@@ -12,59 +12,66 @@ import Mathlib.Tactic.Ring.RingNF
...
 
-The Fermat numbers are a sequence of natural numbers defined as `fermat n = 2^(2^n) + 1`, for all
-natural numbers `n`.
+The Fermat numbers are a sequence of natural numbers defined as `Nat.fermatNumber n = 2^(2^n) + 1`,
+for all natural numbers `n`.
 
 ## Main theorems
 
-- `coprime_fermat_fermat`: two distinct Fermat numbers are coprime.
+- `Nat.coprime_fermatNumberNumber_fermat`: two distinct Fermat numbers are coprime.
```
[the remark this answers is not in the document]
Comment (Vierkantor) [reply Mathlib/NumberTheory/Fermat.lean:20]: Oh no, not again! 😬
```suggestion
- `Nat.coprime_fermatNumber_fermatNumber`: two distinct Fermat numbers are coprime.
```

[the remark this answers is not in the document]
Comment (kbuzzard) [reply Mathlib/NumberTheory/Fermat.lean]: More precisely
```suggestion
- `Nat.coprime_fermatNumber_fermatNumber`: two distinct Fermat numbers are coprime.
```

**Comment by kbuzzard** [comment]
Re "it's too long" -- it will also be never used, so this is fine. There's no big theory of Fermat numbers, I think I would rather argue "this shouldn't even be there" than "this should have a shorter name". Note that `2^2^n+1` is shorter than `Nat.fermat`. My instinct is just to merge this but if people really want to bikeshed about the name then feel free to do it here.

**Comment by kbuzzard** [comment]
I guess one difference is that Mersenne is not famous for anything else! So if I see `mersenne` I can guess what it is.

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**